rawdb/rawdb.go

```diff
@@ -0,0 +1,225 @@
+// Copyright (c) 2024 Hemi Labs, Inc.
+// Use of this source code is governed by the MIT License,
+// which can be found in the LICENSE file.
+
+package rawdb
+
+import (
+	"encoding/binary"
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+	"sync"
+
+	"github.com/juju/loggo"
+	"github.com/syndtr/goleveldb/leveldb"
+)
+
+const (
+	logLevel = "INFO"
+
+	indexDir = "index"
+	dataDir  = "data"
+)
+
+var (
+	log             = loggo.GetLogger("rawdb")
+	lastFilenameKey = []byte("lastfilename")
+)
+
+func init() {
+	loggo.ConfigureLoggers(logLevel)
+}
+
+type RawDB struct {
+	mtx sync.RWMutex
+
+	home string
+
+	maxSize int64
+
+	index *leveldb.DB
+}
+
+func New(home string, maxSize int64) (*RawDB, error) {
+	log.Tracef("New")
+	defer log.Tracef("New exit")
+
+	if maxSize < 4096 {
+		return nil, fmt.Errorf("invalid max size: %v", maxSize)
```

Comment:
nit: could we update this error message to something like:
_invalid max size, must be at least 4096, received %v_